reference/functions/findprocesses.markdown

```diff
@@ -0,0 +1,38 @@
+---
+layout: default
+title: findprocesses
+published: true
+tags: [reference, process functions, functions, findprocesses, process, processes, ps]
+---
+
+[%CFEngine_function_prototype(regex)%]
+
+**Description:** Return the list of files that match the given regular expression `regex`.
+
+This function searches for the given regular expression in the process
+table. Use `.*sherlock.*` to find all the processes that match
+`sherlock`. Use `.*\bsherlock\b.*` to exclude partial matches like
+`sherlock123` (`\b` matches a word boundary).
+
+[%CFEngine_function_attributes(regex)%]
+
+The returned data container is a list of key-value maps. Each one is
+guaranteed to have the key `pid` with the process ID. The key `line`
+will also be available with the raw process table contents.
+
+The process table is usually obtained with something like
+`ps -eo user,pid,ppid,pgid,%cpu,%mem,vsize,ni,rss,stat,nlwp,stime,time,args`
+but the exact data under the `line` key may change per platform and per CFEngine release.
```

Comment:
It's an internal limitation: the code just returns an `ItemList` which is a key-value container. There is no more detail available. We could potentially parse `/proc` and get much more detail, but that's not possible currently.

@estenberg this detail would improve Enterprise reporting as well. With https://github.com/cfengine/core/pull/2245 we have a structured way of parsing data out of `/proc` that should make the task easy. But I don't know if it's possible to transmit data container JSON data to the Enterprise server, so I'm just mentioning it here FYI.